common/src/java/com/zimbra/common/net/SecureProtocolSocketFactoryWrapper.java

```diff
@@ -79,7 +75,18 @@ public Socket createLayeredSocket(Socket socket,  String target, int port, HttpC
             if (context != null) {
                 HttpClientContext clientContext = HttpClientContext.adapt(context);
                 HttpHost  host = clientContext.getTargetHost();
-                return factory.createSocket(host.getHostName(), host.getPort());
+
+                if(host.getPort() == -1) {
```

Comment:
Very similar looking code in 3 places - can they be made into a utility method somewhere?

Reply:
Will do